react-frontend/src/components/Nav/navbar.js

```diff
@@ -1,11 +1,50 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import '../../css/nav.css';
+import { Link, useLocation, useHistory } from 'react-router-dom';
+import '../../font-awesome/css/font-awesome.min.css';
 
 function NavBar() {
+  const history = useHistory();
```

Comment:
I'm kind of curious what this does/is used for

Reply:
When route changes, usehistory is used to fetch the current route name and change the current page. Highlighting in navbar can then watch this state and update itself accordingly